plugin/inlay_hint.py

```diff
@@ -21,7 +21,9 @@ class LspToggleInlayHintsCommand(LspWindowCommand):
 
     def __init__(self, window: sublime.Window) -> None:
         super().__init__(window)
-        window.settings().set('lsp_show_inlay_hints', userprefs().show_inlay_hints)
+        settings = userprefs()
+        default = settings.show_inlay_hints if settings is not None else False
```

Comment:
Note that I put a literal `False` here if `userprefs()` is None, because I assume that we cannot use `sublime.load_settings('LSP.sublime-settings')` before `plugin_loaded()`.

This would only affect the very first window(s) which is/are open when Sublime Text starts.

Reply:
It's kinda not great to assume false when it might not be false.
Could we just not store anything on `init` and only do so when the command is run?

Reply:
Yeah, actually that should work

Reply:
We could also delay reading the default and setting the window setting via using a `sublime.set_timeout` in `__init__`.

But I'm not sure if it is allowed to use `set_timeout` before `plugin_loaded`

Reply:
I'm sure set_timeout is allowed but it doesn't sound very robust.

I don't feel like the case you've mentioned is something that we should need to avoid. It sounds fine to me to follow the global setting until explicitly changing the window setting.

Reply:
I think I would not quite like that, because it might appear very inconsistent to users. If you have multiple windows open and change the value in the settings, then it might be toggled in some of the windows but not in all. And since with my latest commit the window setting will be preserved when initializing, the manual toggle could have happened a long time ago and it would still have an effect here.

So I'd say the toggles should either be a global over all windows (I suggested that earlier when it was implemented, but we decided to not do this), or keep the current behavior and use the setting value only for newly created windows. The latter is also explained in the setting description:
https://github.com/sublimelsp/LSP/blob/8b7afad0527400e113dd6cd0f1f63175bcc599c7/LSP.sublime-settings#L176-L178